enclave_manager/avalon_enclave_manager/wpe/wpe_requester.py

```diff
@@ -41,73 +46,157 @@ def __init__(self, config):
         self._uri_client = HttpJrpcClient(
             config["KMEListener"]["kme_listener_url"])
 
-    def get_unique_verification_key(self, nonce):
+    def get_unique_verification_key(self, verification_key_nonce,
+                                    encryption_key, verifying_key):
         """
         Request wrapper to get a unique id from the KME
 
         Parameters :
-            @param nonce - Random hex nonce generated by this WPE
+            @param verification_key_nonce - Random nonce generated by this WPE
+            @param encryption_key - Encryption key of worker to encrypt request
+            @param verifying_key - Verifying key of the worker
         Returns :
             @returns result - Result received from the KME which includes the
                               public verification key which is supposed to be
                               included in REPORTDATA by the WPE. None, in case
                               of failure.
         """
+
+        workload_id = "kme-uid"
+        in_data = json.dumps({"nonce": verification_key_nonce})
+
+        # Create session key and iv to sign work order request
+        session_key = crypto_utils.generate_key()
+        session_iv = crypto_utils.generate_iv()
+
+        wo_req = self._construct_wo_req(
+            in_data, workload_id, encryption_key, session_key, session_iv)
+
         json_rpc_request = self._get_request_json("GetUniqueVerificationKey")
-        json_rpc_request["params"] = {"nonce": nonce}
+        json_rpc_request["params"] = {"wo_request": wo_req}
```

Comment:
Not sure what are we trying to do here. We have to send  `wo_req`  string obtained from `_construct_wo_req()` to KME listener

`response = self._post_and_get_result(wo_req)`

Reply:
The request always does not go alone. There could be extra fields to be sent as ext_data going along. hence, this structure.

Reply:
Please send ext_data as inData fields. Please use standard TCF format. Do you see any concerns in using standard TC format to send work order request to KME.